server/src/test/java/org/elasticsearch/index/store/FsDirectoryFactoryTests.java

```diff
@@ -156,7 +156,7 @@ private void doTestStoreDirectory(Path tempDir, String typeSettingValue, IndexMo
         Settings settings = settingsBuilder.build();
         IndexSettings indexSettings = IndexSettingsModule.newIndexSettings("foo", settings);
         FsDirectoryFactory service = new FsDirectoryFactory();
-        try (Directory directory = service.newFSDirectory(tempDir, NoLockFactory.INSTANCE, indexSettings)) {
+        try (Directory directory = unwrapDirectory(service.newFSDirectory(tempDir, NoLockFactory.INSTANCE, indexSettings))) {
```

Comment:
maybe we just need to unwrap the directory in the assertions?

Reply:
That makes sense. We do want to test that the wrapper behaves correctly.